lib/dependencies.sh

```diff
@@ -169,6 +171,19 @@ yarn_prune_devdependencies() {
     echo "Skipping because YARN_PRODUCTION is '$YARN_PRODUCTION'"
     meta_set "skipped-prune" "true"
     return 0
+  elif $YARN_2; then
+    cd "$build_dir" || return
+
+    if has_yarn_workspace_plugin_installed "$build_dir"; then
+      meta_set "install-workspace-plugin" "true"
+    else
+      meta_set "install-workspace-plugin" "false"
+      yarn plugin import workspace-tools
+    fi
+
+    rm -rf "$cache_dir"
```

Comment:
I'm sorry I'm probably missing something obvious - why is the cache cleared here in addition to pruning dependencies from the build directory?

Reply:
When pruning with this workspaces plugin, the devDependencies were not getting removed during the prune with a fully installed cache. I thought this was expected behavior since it is an install command, but I went to test the Yarn 1 behavior, and it did remove the devDependencies on the `yarn --production` command. Asking in the Discord channel now.

Reply:
they answered quick - this is the proposed solution for production builds. https://gitlab.com/Larry1123/yarn-contrib/-/tree/master/packages/plugin-production-install

Reply:
I don't want to add another plugin, so removing the directory first seems like the best solution.

Reply:
@danielleadams is it worth adding context in above this line?

Reply:
@hone yeah, I added a comment